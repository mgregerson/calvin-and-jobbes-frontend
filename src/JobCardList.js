import React, { useState } from "react";
import JobCard from "./JobCard";
import "./JobCardList.css";

/** JobCardList
 *
 * Props:
 *      - jobs: [{
 *                   "id": 91,
 *                   "title": "Paramedic",
 *                   "salary": 122000,
 *                   "equity": "0.047",
 *                   "companyName": "Watson Davies",
 *                   "companyHandle": "watson-davies"
 *               } ...]
 *
 * JobList -> JobCardList -> JobCard
 *
 * OR
 *
 * CompanyDetail -> JobCardList -> JobCard
 */

function JobCardList({ jobs }) {
  return (
    <div className="JobCardList">
      {jobs.map((j) => (
        <JobCard
          key={j.id}
          title={j.title}
          salary={j.salary}
          equity={j.equity}
          companyName={j.companyName}
        />
      ))}
    </div>
  );
}

export default JobCardList;
