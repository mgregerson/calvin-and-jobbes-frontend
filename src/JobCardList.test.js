import JobCardList from "./JobCardList";
import { render } from "@testing-library/react";

const TEST_JOBS_LIST = [
  {
    id: 1,
    title: "Test Job 1",
    salary: 50,
    equity: ".1",
    companyHandle: "test-company-1",
    companyName: "Test Company 1",
  },
  {
    id: 2,
    title: "Test Job 2",
    salary: null,
    equity: null,
    companyHandle: "test-company-2",
    companyName: "Test Company 2",
  },
];

describe("JobCardList component", function () {
  test("renders without crashing", function () {
    render(<JobCardList jobs={TEST_JOBS_LIST} />);
  });

  test("matches snapshot", function () {
    const { container } = render(<JobCardList jobs={TEST_JOBS_LIST} />);
    expect(container).toMatchSnapshot();
  });

  test("It creates a new Job component for each job in our prop Jobs List", function () {
    const { container } = render(<JobCardList jobs={TEST_JOBS_LIST} />);
    expect(container.querySelectorAll(".card").length).toEqual(2);
  });
});
