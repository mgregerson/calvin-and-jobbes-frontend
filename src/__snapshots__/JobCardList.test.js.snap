// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobCardList component matches snapshot 1`] = `
<div>
  <div
    class="JobCardList"
  >
    <div
      class="JobCard card my-2"
    >
      <div
        class="JobCard-body"
      >
        <h6
          class="JobCard-title"
        >
          Test Job 1
        </h6>
        <p
          class="JobCard-company-name"
        >
          Test Company 1
        </p>
        <div>
          <small
            class="JobCard-salary"
          >
            Salary: 
            50
          </small>
        </div>
        <div>
          <small
            class="JobCard-equity"
          >
            Equity: 
            .1
          </small>
        </div>
      </div>
    </div>
    <div
      class="JobCard card my-2"
    >
      <div
        class="JobCard-body"
      >
        <h6
          class="JobCard-title"
        >
          Test Job 2
        </h6>
        <p
          class="JobCard-company-name"
        >
          Test Company 2
        </p>
        <div>
          <small
            class="JobCard-salary"
          >
            Unpaid Internship
          </small>
        </div>
      </div>
    </div>
  </div>
</div>
`;
